autogen/agentchat/contrib/capabilities/context_handling.py

```diff
@@ -100,9 +109,11 @@ def truncate_str_to_tokens(text: str, max_tokens: int) -> str:
     Returns:
         Truncated string.
     """
-    truncated_string = ""
-    for char in text:
-        truncated_string += char
-        if token_count_utils.count_token(truncated_string) == max_tokens:
-            break
-    return truncated_string
+
+    tokens = text.split()
+    for token_count in range(max_tokens, 0, -1):
+        truncated_text_tokens = tokens[:token_count]
+        actual_token_count = token_count_utils.count_token(" ".join(truncated_text_tokens))
+        if actual_token_count <= max_tokens:
```

Comment:
Won't this break right away?  actual_token_count will start out smaller than max_tokens

Reply:
This function is calculating tokens by splitting words, which is different than how token_count_utils.count_token() works. This means that the two methods might have slightly different values. This if statement is an extra guard to make sure that the simplistic calculation will never be larger than token_count_utils.count_token. If its larger, than the loop will decrement by string by one token and do the evaluation again. 
If its slightly smaller, I'm ok with that since its below the max threshold.  
Working on some tests for this.